packages/lib/server-only/document/send-document.tsx

```diff
@@ -120,6 +121,29 @@ export const sendDocument = async ({
     Object.assign(document, result);
   }
 
+  const fields = await getFieldsForDocument({
+    documentId: documentId,
+    userId: userId,
+  });
+
+  const fieldsWithSignerEmail = fields.map((field) => ({
+    ...field,
+    signerEmail:
+      document.Recipient.find((recipient) => recipient.id === field.recipientId)?.email ?? '',
+  }));
+
+  const everySignerHasSignature = document?.Recipient.every(
+    (recipient) =>
+      recipient.role !== RecipientRole.SIGNER ||
+      fieldsWithSignerEmail.some(
+        (field) => field.type === 'SIGNATURE' && field.signerEmail === recipient.email,
+      ),
+  );
+
+  if (!everySignerHasSignature) {
+    throw new Error('Some signers have not been assigned a signature field.');
+  }
```

Comment:
This will mean API users will also have to follow these conventions, we should consider if that's something we want @ElTimuro